### PR TITLE
Upgrade dockcross and pybind11

### PR DESCRIPTION
**Pull request by gigony**
Upgrade dockcross image version to support Python 3.10+ for PyPI Wheels.

new Dockcross image: [dockcross/manylinux2014-x64:20230215-2dd9a1c](https://hub.docker.com/r/dockcross/manylinux2014-x64/tags)
new pybind11 version: [v2.10.3](https://github.com/pybind/pybind11/releases/tag/v2.10.3)

Upgrading pybind11 was needed to build Python 3.11+ wheels (compilation error with old pybind11 library).

The latest dockcross image supports gcc10 and `--override` option was needed to install https://developer.download.nvidia.com/compute/cuda/11.0.3/local_installers/cuda_11.0.3_450.51.06_linux.run (to ignore unrecognized compiler versions) in the dockcross image.

```bash
# Build cuda-enabled dev container from `dockcross/manylinux2014-x64:20230215-2dd9a1c` image
./run build_manylinux2014
# Build Python package
./run build_package # using build_python_package is also okay but it wouldn't remove `dist` folder before building.

# wheel file is available in `dist` folder.
```
With this change, wheel file can support Python 3.6~3.11

### CODE REVIEW

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Gigon! 🙏

File: python/cucim/src/cucim/skimage/_vendored/_ndimage_filters.py [hunk 1]
```diff
@@ -1106,6 +1106,9 @@ def _rank_filter(input, get_rank, size=None, footprint=None, output=None,
             sizes = footprint.shape
             has_weights = False
 
+    if not has_weights:
```
Comment (grlee77) [inline python/cucim/src/cucim/skimage/_vendored/_ndimage_filters.py:1109]: Not sure why this shows up here in the diff. The changes in this file are from:
https://github.com/rapidsai/cucim/pull/521

File: python/cucim/src/cucim/skimage/filters/tests/test_median.py [hunk 2]
```diff
@@ -249,3 +250,17 @@ def test_median_preserve_dtype(image, dtype):
 )
 def test_median(img, behavior):
     median(img, behavior=behavior)
+
+
+def test_median_nonsquare():
```
Comment (grlee77) [inline python/cucim/src/cucim/skimage/filters/tests/test_median.py:255]: Not sure why this shows up here in the diff. The changes in this file are from:
https://github.com/rapidsai/cucim/pull/521

**Review by grlee77 — APPROVED** [review]
It looks good to me.

For some reason a couple of changes previously merged in #521 show up here.
**edit**: resolved after merging latest branch-23.04

File: python/cucim/src/cucim/skimage/_vendored/_ndimage_filters.py (hunk 1 above)
Comment (grlee77) [reply python/cucim/src/cucim/skimage/_vendored/_ndimage_filters.py]: resolved now after merging main

**Review by quasiben — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

**Comment by grlee77** [comment]
Thanks @gigony!

**Pull request merged**